src/connection.rs

```diff
@@ -35,17 +62,11 @@ impl fmt::Display for StoreCommand {
     }
 }
 
-impl Connection {
-    /// connect to the memcached server.
-    ///
-    /// Example:
-    ///
-    /// ```rust
-    /// let _ = memcache::Connection::connect("localhost:12345").unwrap();
-    /// ```
-    pub fn connect<A: net::ToSocketAddrs>(addr: A) -> Result<Connection, MemcacheError> {
-        let stream = net::TcpStream::connect(addr)?;
-        return Ok(Connection { reader: io::BufReader::new(stream) });
+impl<C: Read+Write+Sized> Connection<C> {
+
+
+    pub fn from_io(io: C) -> Self {
+        Ok(Connection { reader: io::BufReader::new(io) } )
```

Comment:
Looks like this function is returning a Result, but the function signatures says it returned a Self. And looks like this function is not used now? If so, could you please remove it now.